platform/ui/src/components/Dropdown/Dropdown.jsx

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import { Icon, Typography } from '@ohif/ui';
+
+const Dropdown = ({ titleElement, title, list }) => {
+  const [open, setOpen] = useState(false);
+  const element = useRef(null);
+
+  const renderTitleElement = () => {
+    if (titleElement) {
+      return titleElement;
+    }
+
+    return (
+      <div className="flex">
+        <Typography>{title}</Typography>
+        <Icon name="chevron-down" className="text-white ml-1" />
+      </div>
+    );
+  };
+
+  const toggleList = () => {
+    setOpen(s => !s);
+  };
+
+  const handleClick = e => {
+    if (element.current && !element.current.contains(e.target)) {
+      setOpen(false);
+    }
+  };
+
+  const renderList = () => {
+    const itemsAmount = list.length;
+
+    return (
+      <div
+        className={classnames(
+          'absolute origin-top-right transition duration-300 transform top-100 right-0 mt-2 z-10 bg-primary-dark border border-secondary-main rounded shadow',
+          {
+            'scale-0': !open,
+            'scale-100': open,
+          }
+        )}
+      >
+        {list.map((item, idx) => (
+          <div
+            key={item.title}
+            className={classnames(
+              'flex px-4 py-2 cursor-pointer items-center transition duration-300 hover:bg-secondary-main',
+              {
+                'border-b border-secondary-main': itemsAmount !== idx + 1,
+              }
+            )}
+            onClick={() => {
+              setOpen(false);
+
+              if (item.onClick) {
+                item.onClick();
```

Comment:
You can define a ```const noop = () => {}``` in this jsx file outside the component and use it as default in proptypes? this removes the need to use these types of ifs. (Danny's tip)

Reply:
Good catch! As this is a nested prop of `list`, which is an array, I would have to create a workaround to deal with its default props values. I ended up using an es6 approach (destructuring assignment), I think it's clean and simple. If I had to define a `defaultProps`, another approach would be to split the "list item" into a component to define its default values.

What do you think?

Reply:
I think its fine to just extract the list item into its own component declaration inside this dropdown component and then use defaultProps or destructuring assignment if you wish. Using composing pieces that have semantic names is a lot easier to reason :)

Reply:
I've split the list item into a separated component to set its `defaultProps` instead of destructuring assignment, as we're doing in other components